gems/smithy/lib/smithy/views/client/client.rb

```diff
@@ -60,6 +60,22 @@ def protocols
           @protocols ||= @plan.welds.map(&:protocols).reduce({}, :merge)
         end
 
+        def waiters
+          waiters = Views::Client::Waiters.new(@plan).waiters
+          lines = ['{']
+          waiters.each_with_index do |waiter, i|
+            line = '  ' + waiter.name.underscore + ': Waiters::' + waiter.name + ','
+            line.chomp!(',') if i == waiters.length - 1
```

Comment:
Check out the operation examples code. You shouldn't need to keep track of index I think. You can just run the loop and always chomp comma.